app/models/course.rb

```diff
@@ -162,4 +165,15 @@ def open!
       opened_on_apply_at: Time.zone.now,
     )
   end
+
+private
+
+  def touch_application_choices_and_forms
+    return if saved_changes['start_date'].blank?
+
+    ActiveRecord::Base.transaction do
+      application_choices.touch_all
+      ApplicationForm.where(application_choices: application_choices).touch_all
```

Comment:
Doesn't this https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/app/models/application_choice.rb#L7 touch the form when the choices are touched?

Reply:
No because `touch_all` doesn't trigger callbacks.